File: package/ios/expo/views/SmileIDExpoSmartSelfieEnrollmentView.swift
import ExpoModulesCore
import UIKit
import SwiftUI
import SmileID

/**
 * Expo implementation of SmartSelfieEnrollment using the new architecture
 */
class SmileIDExpoSmartSelfieEnrollmentView: SmileIDExpoView {
    
    var allowNewEnroll: Bool = true
    
    override func renderContent() {
        let config = createConfig()
        
        // Use SmileIDExtReactNative to create the view
        let uiView = SmileID.RNSmartSelfieEnrollment(
            config: config,
            onResult: { [weak self] result in
                self?.handleResultCallback(result)
            }
        )
        
        // Use the base class helper to set the UIView
        setUIView(uiView)
    }
    
    private func createConfig() -> SmileIDViewConfig {
        var config = SmileIDViewConfig()
        config.userId = userId
        config.jobId = jobId
        config.allowAgentMode = allowAgentMode
        config.allowNewEnroll = allowNewEnroll
        config.showInstructions = showInstructions
        config.skipApiSubmission = skipApiSubmission
        config.showAttribution = showAttribution
        config.extraPartnerParams = extraPartnerParams
        return config
    }
}

File: package/android/src/expo/com/smileidentity/expo/views/SmileIDExpoSmartSelfieEnrollmentView.kt
package com.smileidentity.expo.views

import android.content.Context
import androidx.compose.runtime.Composable
import com.smileidentity.SmileID
import com.smileidentity.shared.RNSmartSelfieEnrollment
import com.smileidentity.shared.SmileIDViewConfig
import expo.modules.kotlin.AppContext

/**
 * Expo implementation of SmartSelfieEnrollment using the new architecture
 */
class SmileIDExpoSmartSelfieEnrollmentView(
    context: Context,
    appContext: AppContext
) : SmileIDExpoView(context, appContext) {

    var allowNewEnroll: Boolean = false

    override fun renderContent() {
        setContentWithTheme {
            RenderSmartSelfieEnrollmentContent()
        }
    }

    @Composable
    private fun RenderSmartSelfieEnrollmentContent() {
        val config = SmileIDViewConfig(
            userId = userId,
            jobId = jobId,
            allowAgentMode = allowAgentMode ?: false,
            showInstructions = showInstructions,
            skipApiSubmission = skipApiSubmission,
            showAttribution = showAttribution,
            extraPartnerParams = extraPartnerParams,
            allowNewEnroll = allowNewEnroll
        )

        SmileID.RNSmartSelfieEnrollment(
            config = config,
            onResult = { result ->
                handleResultCallback(result)
            }
        )
    }
}